<launch>
	<!-- This needed due to bug in rviz visualization on melodic. When bug is fixed line below could be deleted -->
	<env name="LC_NUMERIC" value="en_US.UTF-8" />

	<arg name="model" default="$(find porter_robot_model)/xacro/model.xacro"/>

	<arg name="rvizconfig" default="$(find porter_robot_simulation)/rviz/default.rviz" />
	
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" />
	
	<param name="porter/robot_description" command="xacro $(arg model)" />

	<node pkg="odrive_ros" type="odrive_node" name="odrive" ns="porter" 
            output="screen" respawn="false">
        <param name="wheel_track" value="0.285"/>
        <param name="tyre_circumference" value="0.79796453401"/>
        
        <param name="connect_on_startup"   value="true"/>
        <param name="calibrate_on_startup" value="true"/>
        <param name="engage_on_startup"    value="true"/>
        
        <param name="publish_odom"         value="true"/>
        <param name="publish_odom_tf"      value="true"/>
		
		<param name="~base_frame"		   value="chasiss"/>		

		<remap from="/cmd_vel" to="/porter/cmd_vel"/>
		<remap from="/odrive/joint_states" to="/porter/joint_states"/>
    </node>
	
	<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" ns="porter">
    	<param name="publish_frequency" type="double" value="60.0" />
  	</node>
	
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" ns="porter">
              <param name="use_gui" value="False"/>
    </node>
    
    <node name="rqt_robot_steering" pkg="rqt_robot_steering" type="rqt_robot_steering">
		<param name="default_topic" value="/porter/cmd_vel"/>
	</node>

</launch>
